src/serac/physics/solid_mechanics.hpp

```diff
@@ -915,14 +980,10 @@ class SolidMechanics<order, dim, Parameters<parameter_space...>, std::integer_se
   void addBodyForce(DependsOn<active_parameters...>, BodyForceType body_force,
                     const std::optional<Domain>& optional_domain = std::nullopt)
   {
-    Domain domain = (optional_domain) ? *optional_domain : EntireDomain(mesh_);
-
-    residual_->AddDomainIntegral(
-        Dimension<dim>{}, DependsOn<0, 1, active_parameters + NUM_STATE_VARS...>{},
-        [body_force](double t, auto X, auto /* displacement */, auto /* acceleration */, auto... params) {
-          return serac::tuple{-1.0 * body_force(get<VALUE>(X), t, params...), zero{}};
-        },
-        domain);
+    Domain                            domain = (optional_domain) ? *optional_domain : EntireDomain(mesh_);
+    BodyForceIntegrand<BodyForceType> force_integrand(body_force);
+    residual_->AddDomainIntegral(Dimension<dim>{}, DependsOn<0, 1, active_parameters + NUM_STATE_VARS...>{},
+                                 std::move(force_integrand), domain);
```

Comment:
Just construct this object in place instead of constructing then calling std::move.